app/types/index.ts

```diff
@@ -1,12 +1,33 @@
+import type { Server } from "socket.io";
+import type { Socket } from "socket.io-client";
 import type { Field } from "@/utils/game";
 
 export type ServerToClientEvents = {
   update(state: Field["plots"]): void;
   clientsCount(count: number): void;
   exposedPercent(count: number): void;
+  session: ({ sessionID }: { sessionID: string }) => void;
```

Comment:
since we're only concerned with the session, we can drop the use of an object here (also, I'm personally partial to using `sessionId` over `sessionID`)
```suggestion
  session: (sessionId: string) => void;
```